server/lorax_server/layers/fp8.py

```diff
@@ -34,9 +34,9 @@ def __init__(
         super().__init__()
         self.dtype = weight.dtype
         self.qweight = weight.t()
-        self.weight_scale = weight_scale.view(1, -1).contiguous()
+        self.weight_scale = weight_scale.view(1, -1).contiguous().float()
         self.qbias = bias if bias is not None else None
-        self.input_scale = input_scale
+        self.input_scale = input_scale.float() if input_scale else input_scale
```

Comment:
Is this a check for None? If so, can we be more explicit in that check:

x.float() if x is not None else None